src/ecs/commands/runCommandInContainer.ts

```diff
@@ -91,3 +119,25 @@ export async function runCommandInContainer(
         status?.dispose()
     }
 }
+
+async function checkRequiredPermissions(node: EcsContainerNode): Promise<boolean> {
+    if (node.taskRoleArn === undefined) {
+        return false
+    }
+    const iamClient = globals.toolkitClientBuilder.createIamClient(node.ecs.regionCode)
+    const permissionResponse = (
+        await iamClient.simulatePrincipalPolicy({
```

Comment:
What happens if the role doesn't have `simulatePrincipalPolicy` permissions? In that case it may be best to not return false.

Reply:
I'd also log something about not being able to simulate permissions.
It's overkill here but instead you could return something other than a boolean so callers can craft a specific message depending on why permissions weren't enough.

Reply:
I'm changing this logic up a bit. This will now _attempt_ to check if the SSM permissions are missing, but not block if they are not able to call `simulatePrincipalPolicy`. I will log any errors that come out of that attempt.